vespamalloc/src/tests/test1/new_test.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#include <vespa/vespalib/testkit/testapp.h>
+#include <vespa/log/log.h>
+
+LOG_SETUP("new_test");
+
+void cmp(const void *a, const void *b) {
+    EXPECT_EQUAL(a, b);
+}
+void cmp(const void *base, size_t offset, const void *p) {
+    cmp((static_cast<const char *>(base) + offset), p);
+}
+
+TEST("verify new with normal alignment") {
+    struct S {
+        int a;
+        long b;
+        int c;
+    };
+    static_assert(sizeof(S) == 24, "sizeof(S) == 16");
+    auto s = std::make_unique<S>();
+    cmp(s.get(), &s->a);
+    cmp(s.get(), 8, &s->b);
+    cmp(s.get(), 16, &s->c);
```

Comment:
For these tests, should we additionally check the returned pointer alignment itself? The struct _field_ alignments should be up to the compiler to get right. Might be enough to add a templated checker function that verifies  `(uintptr_t(s.get()) % alignof(S)) == 0`?

Reply:
Done, and I also remembered that only vespamalloc is naturally aligned. vespamallocd and vespamallocdst is not as they add preamble and postamble. So they need to be propagated.